src/map/map.ts

```diff
@@ -1018,6 +1018,13 @@ export class PropertiesMap {
         layout.coloraxis.colorbar.len = this._colorbarLen();
         layout.coloraxis.showscale = this._options.hasColors();
 
+        // Set ranges for the axes
+        const setAxisRange = (min: number, max: number) => {
+            return min !== 0 && max !== 0 ? [min, max] : undefined;
+        };
+        layout.xaxis.range = setAxisRange(this._options.x.min.value, this._options.x.max.value);
+        layout.yaxis.range = setAxisRange(this._options.y.min.value, this._options.y.max.value);
```

Comment:
Do the other axis-looking properties have the right range? These should be the z axis and color axis.

Reply:
Yes, indeed, it could be fixed for the z-axis & color as well